File: apps/android/app/src/main/java/com/alafteknoloji/pagapp/services/VerificationService.kt
package com.alafteknoloji.pagapp.services

import com.alafteknoloji.pagapp.models.PAGQuestion
import com.alafteknoloji.pagapp.models.PAGQuestionOption
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.flow.asStateFlow
import org.json.JSONArray
import org.json.JSONObject

data class PendingVerificationSurveyData(
    val assignmentId: String,
    val verificationSurveyId: String,
    val masterSurveyId: String,
    val masterSurveyTitle: String,
    val title: String,
    val description: String,
    val rewardSummary: String,
    val questionCount: Int,
    val questions: List<PAGQuestion>
)

class VerificationService private constructor() {

    private val _pendingVerification = MutableStateFlow<PendingVerificationSurveyData?>(null)
    val pendingVerification: StateFlow<PendingVerificationSurveyData?> = _pendingVerification.asStateFlow()

    private val _isLoading = MutableStateFlow(false)
    val isLoading: StateFlow<Boolean> = _isLoading.asStateFlow()

    private val _isSubmitting = MutableStateFlow(false)
    val isSubmitting: StateFlow<Boolean> = _isSubmitting.asStateFlow()

    private val _errorMessage = MutableStateFlow<String?>(null)
    val errorMessage: StateFlow<String?> = _errorMessage.asStateFlow()

    suspend fun checkPendingVerification() {
        _isLoading.value = false
        _pendingVerification.value = null
    }

    suspend fun submitVerificationAnswer(
        surveyId: String,
        questionId: String,
        optionId: String
    ): Boolean {
        _isSubmitting.value = true
        _errorMessage.value = null

        return try {
            val answersArray = JSONArray().apply {
                put(JSONObject().apply {
                    put("questionId", questionId)
                    put("optionId", optionId)
                })
            }
            val body = JSONObject().apply {
                put("answers", answersArray)
            }
            val apiRes = PAGApiClient.post("/surveys/$surveyId/submit", body)
            val success = apiRes != null && apiRes.optBoolean("success")
            if (success) {
                _pendingVerification.value = null
            }
            _isSubmitting.value = false
            success
        } catch (e: Exception) {
            _isSubmitting.value = false
            _errorMessage.value = e.localizedMessage
            false
        }
    }

    fun dismissForNow() {
        _pendingVerification.value = null
    }

    companion object {
        val shared = VerificationService()
    }
}

File: apps/ios/PagApp/PagApp/Services/VerificationService.swift
import Foundation
import Combine

public struct PendingVerificationSurvey: Identifiable {
    public var id: String { assignmentId }
    public let assignmentId: String
    public let verificationSurveyId: String
    public let masterSurveyId: String
    public let masterSurveyTitle: String
    public let title: String
    public let description: String
    public let rewardSummary: String
    public let questionCount: Int
    public let questions: [PAGQuestion]
}

@MainActor
public class VerificationService: ObservableObject {
    public static let shared = VerificationService()
    
    @Published public private(set) var pendingVerification: PendingVerificationSurvey? = nil
    @Published public private(set) var isLoading: Bool = false
    @Published public private(set) var isSubmitting: Bool = false
    @Published public private(set) var errorMessage: String? = nil
    
    private init() {}
    
    public func checkPendingVerification() async {
        self.isLoading = false
        self.pendingVerification = nil
    }
    
    public func submitVerificationAnswer(surveyId: String, questionId: String, optionId: String) async throws -> Bool {
        self.isSubmitting = true
        self.errorMessage = nil
        
        do {
            let payload: [String: Any] = [
                "answers": [
                    ["questionId": questionId, "optionId": optionId]
                ]
            ]
            let result = try await PAGApiClient.shared.post(endpoint: "/surveys/\(surveyId)/submit", body: payload)
            guard let success = result["success"] as? Bool, success else {
                throw NSError(domain: "VerificationService", code: 500, userInfo: [NSLocalizedDescriptionKey: "Yanıt kaydedilemedi."])
            }
            
            self.pendingVerification = nil
            self.isSubmitting = false
            return true
        } catch {
            self.isSubmitting = false
            self.errorMessage = error.localizedDescription
            throw error
        }
    }
    
    public func dismissForNow() {
        self.pendingVerification = nil
    }
}
